### DISCUSSION TITLE
Presence in Mastodon

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-core#9940] [Q&A]
Micronaut foundation thinks we should create an official account for the Micronaut Framework in [Mastodon](https://www.youtube.com/watch?v=IPSbNdBmWKE). 

Options: 

A) Host a Mastodon instance. 
B) Create an account in an existing instance. 

## Own instance

Examples of option A. 

https://tapbots.social/
https://kotlin.social

We could start with an instance with one account for the framework. 

Advantages: complete control. We don't rely on an instance managed by anyone else being decommissioned or banned. 

Disadvantages: We have to host it and run it. 

## Third-party instance

We would need to choose an instance. 

Some candidates: 

[**foojay.social**](https://foojay.social/about):

> This Foojay.social Mastodon server is aimed at anyone active in the OpenJDK, Java, JavaFX, JVM, and related communities.

[**jvm.social**](https://jvm.social/about)

> This is a mastodon server devoted to the JVM (Java) community. As such, it should be useful for developers using that platform and ideally it should be used to talk about our beloved programming language (Java, Kotlin, Scala, Groovy, Clojure... you name it). This is not a fixed rule: you won't be banned from the server if you talk about other topic, but a technical profile is desirable to have a interesting local timeline.

[**fosstodon.social**](https://fosstodon.social/about)

> Fosstodon is a Mastodon instance that is open to anyone who is interested in technology; particularly free & open source software.

Open-Source projects such as [Quarkus](https://fosstodon.org/@quarkusio) or [Apache Groovy](https://fosstodon.org/@apachegroovy) have profiles in Fosstodon. 

Thoughts?

**Comment by sdelamo** [comment]
I would prefer to host our instance if someone from the community offered to cover the costs of hosting the instance. They could become a [tools and infrastructure sponsor](https://micronaut.io/foundation/sponsors/) of the Micronaut Foundation.

**Comment by musketyr** [comment]
join the one where there are already the friendly projects. if you run your own server, it will be more difficult to find you because as far as I know the user searches only works within one server. if fosstodon.social is the server where people go to find a good OSS projects, I would go there.

**Comment by jagedn** [comment]
I think is better to join one of the mentioned instances as Mastodon (in fact Fediverse) is more related to interact with people and friends. If you run your own instance you'll need to federated it with many others to gain visibility 

Also, as a personal opinion, a Mastodon/Fediverse account is interesting if you have plans to interact with users not only to publish news or releases, but this is a personal opinion.

**Reply by sdelamo** [reply]
I anticipate the account to be mostly a way to announce releases, content, etc. I don't expect a lot of interaction with users.

**Comment by timyates** [comment]
I agree with using [fosstodon.social](https://fosstodon.social/about). Running and maintaining an instance doesn't feel like a good use of our resources.

Fosstodon is invite only, but hopefully we can get an invite off the Groovy project?

**Reply by paulk-asert** [reply]
I can arrange the invite if you go that way.

**Comment by roamingthings** [comment]
I think it's a good idea to join an existing Mastodon instance. I can imagine that it's a greater effort for managing an own instance. In addition, it's not clear today what future liability or regulation will mean for content and user moderation.

**Reply by sdelamo** [reply]
I am not advocating for an instance where we allow profiles from community members. I am advocating for an instance with 2 or 3 users. Initially one `micronautfw`. I don't anticipate any issues with content and/or user moderation. Because we will not open the doors for content or users.

**Comment by sdelamo** [comment]
https://jvm.social/@jmiguel/111183442591129146

> @sdelamo@jvm.social Sure!. I've been running the jvm.social for about 10 months now.  It runs on a small Hetznet hosting machine (2 vcpus, 8GB RAM, 80GB drive) that cost less than 12€/month. 
> 
> I store all images on AWS S3 so disk space can't break the system unexpectedly. Add ~8€/month.
> 
> All invoices at https://opencollective.com/jvm-social/expenses 
> 
> Add domain name and pretty much that's it.
> 
> I also run my own mono-user server on a smaller free instance. 
> 
> Mastodon requirements are not very high: 3 ruby processes + postgres
> 
> https://jvm.social/@jmiguel/111183442591129146

**Comment by sdelamo** [comment]
Helidon is in https://mastodon.social/@helidon

**Comment by sdelamo** [comment]
We are going to create a mastodon account in fosstodon.

**Comment by sdelamo** [comment]
should we use in mastodon `https://fosstodon.org/@Micronautfw` or `https://fosstodon.org/@Micronaut` as handle ?. 

Please use thumbsup for `Micronautfw` or thumbsdown for `Micronaut`

**Comment by sdelamo** [comment]
we have a profile https://fosstodon.org/@micronaut
